trackintel/preprocessing/positionfixes.py

```diff
@@ -164,7 +159,11 @@ def generate_staypoints(
     # user_id of sp should be the same as ret_pfs
     sp["user_id"] = sp["user_id"].astype(pfs["user_id"].dtype)
 
-    return pfs, sp
+    if len(sp) <= 0:
+        warnings.warn("No staypoints can be generated, returning empty sp.")
+        return pfs, sp
+
+    return pfs, Staypoints(sp)
```

Comment:
Should we write `Positionfixes(pfs)` here? (Similar to what is changed in `trackintel/preprocessing/staypoints.py`). I think `pfs` shall be already valid, so rechecking it is redundant, but it would be good to be consistent across functions

Reply:
The idea was not changing the input type of pfs if pfs is a GeoDataFrame. But as we checked `Positionfixes.valdiate(pfs)` I would be okay with either. What do would you expect as a user using this function?